arch/riscv/core/smp.c

```diff
@@ -18,6 +19,10 @@ volatile struct {
 volatile uintptr_t riscv_cpu_wake_flag;
 volatile void *riscv_cpu_sp;
 
+#ifdef CONFIG_PM_CPU_OPS
+extern void __start(void);
```

Comment:
should this be `CONFIG_KERNEL_ENTRY`?

Reply:
Are you referring to the above #ifdef check ? or function declaration ?

If its for function declaration, 
I am not sure, if we have any helpers to declare function like: 
extern void CONFIG_KERNEL_ENTRY(void);

Reply:
uhm, I see. This should really be fixed somehow, it is really useless to have `CONFIG_KERNEL_ENTRY` and not being able to use it in C code (but only in the linker script).